storage/file.go

```diff
@@ -92,40 +292,176 @@ func pathForFileShare(name string) string {
 // See https://msdn.microsoft.com/en-us/library/azure/dd179352.aspx
 func (f FileServiceClient) ListShares(params ListSharesParameters) (ShareListResponse, error) {
 	q := mergeParams(params.getParameters(), url.Values{"comp": {"list"}})
-	uri := f.client.getEndpoint(fileServiceName, "", q)
-	headers := f.client.getStandardHeaders()
 
 	var out ShareListResponse
-	resp, err := f.client.exec("GET", uri, headers, nil)
+	resp, err := f.listContent("", q, nil)
 	if err != nil {
 		return out, err
 	}
-	defer resp.body.Close()
 
+	defer resp.body.Close()
 	err = xmlUnmarshal(resp.body, &out)
 	return out, err
 }
 
-// CreateShare operation creates a new share under the specified account. If the
-// share with the same name already exists, the operation fails.
+// retrieves directory or share content
+func (f FileServiceClient) listContent(path string, params url.Values, extraHeaders map[string]string) (*storageResponse, error) {
+	if err := f.checkForStorageEmulator(); err != nil {
+		return nil, err
+	}
+
+	uri := f.client.getEndpoint(fileServiceName, path, params)
+	headers := mergeHeaders(f.client.getStandardHeaders(), extraHeaders)
+
+	resp, err := f.client.exec(http.MethodGet, uri, headers, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = checkRespCode(resp.statusCode, []int{http.StatusOK}); err != nil {
+		resp.body.Close()
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+// CreateDirectory operation creates a new directory in the specified share. If a
+// directory with the same name already exists, the operation fails.
 //
-// See https://msdn.microsoft.com/en-us/library/azure/dn167008.aspx
-func (f FileServiceClient) CreateShare(name string) error {
-	resp, err := f.createShare(name)
+// See https://msdn.microsoft.com/en-us/library/azure/dn166993.aspx
+func (f FileServiceClient) CreateDirectory(path string) error {
```

Comment:
CreateFile and CreateDirectory were modeled from CreateShare which doesn't allow one to specify metadata at the time of creation.  If we want to fix this for files and dirs I think we should also fix it for shares even though it's a breaking change.  What do you think?